Python/Product/ExternalProfilerDriver/ProcessAsyncRunner.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace Microsoft.PythonTools.Profiling.ExternalProfilerDriver
+{
+    class ProcessAsyncRunner
+    {
+        static public Task<Process> Run(string execfname,
+                                        string args = null,
+                                        CancellationToken cancellationToken = default(CancellationToken),
+                                        IProgress<ProcessProgressDataload> progress = null
+                                        )
+        {
+            TaskCompletionSource<Process> taskCS = new TaskCompletionSource<Process>();
+
+            bool stdoutToProgress = (progress != null);
+
+            ProcessStartInfo psi = new ProcessStartInfo(execfname)
+            {
+                UseShellExecute = false,
+                Arguments = args,
+                CreateNoWindow = false,
+                RedirectStandardOutput = stdoutToProgress
+            };
+            psi.EnvironmentVariables["AMPLXE_EXPERIMENTAL"] = "time-cl";
+
+            Process process = new Process
+            {
+                StartInfo = psi,
+                EnableRaisingEvents = true
+            };
+
+            process.Exited += (sender, localEventArgs) =>
```

Comment:
Since you're using redirected output, you should use its event when `e.Data == null` to detect when the process exits. 

The Exited event may be raised before the output is fully read.

Reply:
I believe we also need error stream to be drained as well.